File: AndroidPOS/app/src/main/java/com/team/androidpos/model/dao/CudDao.java
package com.team.androidpos.model.dao;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Update;

public interface CudDao<E> {

    @Insert
    void insert(E entity);

    @Update
    void update(E entity);

    @Delete
    void delete(E entity);

}

File: AttendanceKt/app/src/main/java/com/team/attendancekt/model/dao/CudDao.kt
package com.team.attendancekt.model.dao

import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Update

interface CudDao<E> {

    @Insert
    fun insert(entity: E)

    @Update
    fun update(entity: E)

    @Delete
    fun delete(entity: E)

}